go.mod

```diff
@@ -4,9 +4,9 @@ go 1.21
 
 require (
 	emperror.dev/errors v0.8.1
-	github.com/bank-vaults/internal v0.1.0
+	github.com/bank-vaults/internal v0.2.0
 	github.com/bank-vaults/vault-sdk v0.9.0
-	github.com/hashicorp/vault/api v1.9.2
```

Comment:
This PR also updates the Vault API version. Is this a required version bump and are there any breaking changes that should be addressed? I went through the API release notes briefly (feel free to double check), but could not find anything relevant.

Reply:
In other BV packages it is already bumped to 1.10.0, internal included, I guess that is why it got bumped here as well with `go mod tidy`. I also didn't notice anything breaking so I left it here...